### PR TITLE
Create a button in palette for all schema

### PR DESCRIPTION
**Pull request by marthacryan**
Automatically detects all schema and creates a button to open the widget for a given schema.
image

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by kevin-bates** [comment]
Hmm.  The plurality is off.  Since namespace names are plural and schema names are singular, I wonder if the namespace name is what is wanted here anyway.  Since 2 of the 3 are single-schema namespaces, they just work.  The other would should KFP and AIrflow "runtimes" leaving the headings with...

Show Code Snippets
Show Runtime Images
Show Runtimes

Actually, the namespace names are lowercase and values like "code-snippets", "runtime-images" and "runtimes".  So this is another great example of promoting namespaces to full objects where they too could have display_name values - leading to what I showed above (with proper case and spacing).  Until then, I don't think this is going to fly, unfortunately.

**Comment by marthacryan** [comment]
@kevin-bates What if we add 'Widget' after each of them? So "Show Code Snippet Widget" or "Show Runtime Image Widget"

**Comment by kevin-bates** [comment]
To show a "widget" sounds odd to me.  

Also, I think we should forgo using `display_name` from the **schema** because its more of an application property _defined in the schema_.  Instead, we added `title` everywhere and that's what should be used here instead.  When dealing with metadata _instances_ you would reference `display_name`.

**Comment by marthacryan** [comment]
@kevin-bates Yeah good point - maybe "browser"?

File: packages/metadata/src/index.ts [hunk 1]
```diff
@@ -155,6 +157,26 @@ const extension: JupyterFrontEndPlugin<void> = {
...
+    for (const schema of schemas) {
+      let icon = 'ui-components:text-editor';
+      if (schema.uihints && schema.uihints.icon) {
+        icon = schema.uihints.icon;
+      }
+      palette.addItem({
+        command: commandIDs.openMetadata,
+        args: {
+          label: `Show ${schema.display_name}`,
+          display_name: schema.display_name,
```
Comment (kevin-bates) [inline packages/metadata/src/index.ts:172]: We should switch to using `title` from the schema file.
```suggestion
          label: `Show ${schema.title}`,
          display_name: schema.title,
```

**Comment by marthacryan** [comment]
How's this look? I made it so that it uses the `title` by default but if there's a `uihints.title` it'll use that instead. 
image

**Review by karlaspuldaro — APPROVED** [review]
LGTM :)

**Comment by marthacryan** [comment]
Ok while working with this I figured it might be good to also implement that "Close tab" ability for metadata viewers in the left side pane since they can really add up and now it's easy to reopen any of them. Just pushed those changes but if you don't think it's good with this PR I can undo that. Thoughts?

**Comment by marthacryan** [comment]
Just added a small change so that it won't show the "Close tab" option for code snippets or runtimes (because they're supposed to be more permanent). It was causing a small bug of creating a generic metadata widget for code snippets (which doesn't include the coding languages in the display and doesn't include the insert / copy buttons).

**Comment by marthacryan** [comment]
So here's what the context menu looks like for code snippets / runtimes:
image

And here's the context menu for any other metadata schema present:
image

**Review by karlaspuldaro — APPROVED** [review]
Nice adding a close tab option. Re-ran this locally and looks good.

**Review by lresende — APPROVED** [review]
LGTM - Tested locally and all looks good.

**Pull request merged**